geowebcache/core/src/test/java/org/geowebcache/blobstore/file/FileBlobStoreComformanceTest.java

```diff
@@ -27,4 +35,69 @@
     public void createTestUnit() throws Exception {
         this.store = new FileBlobStore(temp.getRoot().getAbsolutePath());
     }
+
+    @Test
+    public void testMetadataWithPointInKey() throws Exception {
+        assertThat(store.getLayerMetadata("testLayer", "test.Key"), nullValue());
+        store.putLayerMetadata("testLayer", "test.Key", "testValue");
+        assertThat(store.getLayerMetadata("testLayer", "test.Key"), equalTo("testValue"));
+    }
+
+    @Test
+    public void testConcurrentMetadataWithPointInKey() throws InterruptedException {
+        assertThat(store.getLayerMetadata("testLayer", "test.Key"), nullValue());
+        int numberOfThreads = 2;
+        ExecutorService service = Executors.newFixedThreadPool(numberOfThreads);
+        CountDownLatch latch = new CountDownLatch(numberOfThreads);
+        for (int i = 0; i < numberOfThreads; i++) {
+            final int key = i;
+            service.submit(
+                    () -> {
+                        // Sleep thread randomly before adding information into metadata file.
+                        try {
+                            Thread.sleep(Math.round((Math.random() * 500) + 1));
+                        } catch (InterruptedException e) {
+                            // Handle exception
+                        }
+                        store.putLayerMetadata(
+                                "testLayer", "test.Key." + String.valueOf(key), "testValue");
+                        latch.countDown();
+                    });
+        }
+        latch.await();
+        assertThat(store.getLayerMetadata("testLayer", "test.Key.1"), equalTo("testValue"));
+        assertThat(store.getLayerMetadata("testLayer", "test.Key.0"), equalTo("testValue"));
+    }
+
+    @Test
+    public void testConcurrentMassiveMetadataKeys() throws InterruptedException {
```

Comment:
This could take a long time, useful to have for development and eventual double checks, but I'd mark it as "@Ignore" long term, or at least make sure it does not take too much time (seconds is fine, minutes is not) and cannot slam down the machine (50 threads on a single core computer is too much, on a 128 cores one is not enough, what about using Runtime to get the number of cores, and make the test use 2-4 times more than available?).